expression/builtin_info.go

```diff
@@ -383,8 +383,56 @@ type benchmarkFunctionClass struct {
 	baseFunctionClass
 }
 
+// getFunction See https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_benchmark
 func (c *benchmarkFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "BENCHMARK")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// Syntax: BENCHMARK(loop_count, expression)
+	// Now that eval result of input arg is useless, we can define by the same eval type of input arg here.
+	sameEvalType := args[1].GetType().EvalType()
```

Comment:
[`Only scalar expressions can be used`](https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_benchmark), so the type of `args[1]` must be taken into account.

`select benchmark(100000, (select * from t))` is valid in this PR, while it results in an error in mysql.

Reply:
@qw4990 I think MySQL supports subquery in `BENCHMARK()`, the full paragraph in document is:

> Only scalar expressions can be used. Although the expression can be a subquery, it must return a single column and at most a single row. For example, BENCHMARK(10, (SELECT * FROM t)) will fail if the table t has more than one column or more than one row.

So as long as the query result is no more than 1 column and 1 row, `BENCHMARK()` will execute with no error. Here's the testcase running on my local MySQL 5.7.25, and the same testcase of TiDB is in [integration_test.go](https://github.com/pingcap/tidb/pull/9252/files#diff-dd33813624d57c13ba9794ebc82005c4R2501) of this PR.
```
mysql> SHOW VARIABLES LIKE "%version%";
+-------------------------+------------------------------+
| Variable_name           | Value                        |
+-------------------------+------------------------------+
| innodb_version          | 5.7.25                       |
| protocol_version        | 10                           |
| slave_type_conversions  |                              |
| tls_version             | TLSv1,TLSv1.1                |
| version                 | 5.7.25                       |
| version_comment         | MySQL Community Server (GPL) |
| version_compile_machine | x86_64                       |
| version_compile_os      | Linux                        |
+-------------------------+------------------------------+
8 rows in set (0.00 sec)

mysql> drop table if exists t;
Query OK, 0 rows affected (0.02 sec)

mysql> create table t (a int, b int);
Query OK, 0 rows affected (0.07 sec)

mysql> insert t values (1, 2);
Query OK, 1 row affected (0.00 sec)

mysql> select benchmark(10, (select a from t));
+----------------------------------+
| benchmark(10, (select a from t)) |
+----------------------------------+
|                                0 |
+----------------------------------+
1 row in set (0.00 sec)

mysql> select benchmark(10, (select * from t));
ERROR 1241 (21000): Operand should contain 1 column(s)
mysql> insert t values (3, 4);
Query OK, 1 row affected (0.01 sec)

mysql> select benchmark(10, (select a from t));
ERROR 1242 (21000): Subquery returns more than 1 row
```